storage/aptosdb/src/state_merkle_db.rs

```diff
@@ -182,7 +183,12 @@ impl TreeReader<StateKey> for StateMerkleDb {
         let mut iter = self.iter::<JellyfishMerkleNodeSchema>(Default::default())?;
         iter.seek_to_first();
         let version = match iter.next().transpose()? {
-            Some((node_key, _node)) => node_key.version(),
+            Some((node_key, node)) => {
+                if node.node_type() == NodeType::Null {
+                    return Ok(None);
+                }
+                node_key.version()
+            }
```

Comment:
For my own understanding, why do we need to change this? Is this `fix the restore of empty tree` you mentioned?

Reply:
yes

Reply:
Otherwise another test will fail.